drivers: pwm: sam: compilation failure for sam_v71b_xult

**Describe the bug**
When building for the sam_v71b_xult with the `pwm_sam` driver enabled, the build fails.

**To Reproduce**
Steps to reproduce the behavior:
1. `west build -p=auto -b sam_v71b_xult zephyr/samples/hello_world/ -t menuconfig`
2. Enable the `pwm_sam` driver
3. `west build -p=auto -b sam_v71b_xult zephyr/samples/hello_world/`

**Expected behavior**
The sample builds with enabled pwm driver.

**Impact**
Can not use the `pwm_sam` driver for the sam_v71b_xult board.

**Logs and console output**
Relevant build failure message:
```
/home/christoph/git/zephyr/drivers/pwm/pwm_sam.c:64:5: error: 'Pwm' has no member named 'PWM_CH_NUM'
   64 |  pwm->PWM_CH_NUM[ch].PWM_CMR = PWM_CMR_CPRE_CLKA_Val;
      |     ^~
/home/christoph/git/zephyr/drivers/pwm/pwm_sam.c:69:5: error: 'Pwm' has no member named 'PWM_CH_NUM'
   69 |  pwm->PWM_CH_NUM[ch].PWM_CPRDUPD = period_cycles;
      |     ^~
/home/christoph/git/zephyr/drivers/pwm/pwm_sam.c:70:5: error: 'Pwm' has no member named 'PWM_CH_NUM'
   70 |  pwm->PWM_CH_NUM[ch].PWM_CDTYUPD = pulse_cycles;
```

**Environment (please complete the following information):**
 - OS: Debian 11 Testing
 - Zephyr SDK
 - Commit 4113a43bfff48f58bf5300163e2efbe046f1e5f8 (current main at filing of this issue)


**Additional context**
This is very likely related to the Atmel SoC HAL which defines these structs. There are very similar defines available, which define no struct, but bit masks for the corresponding registers. I am currently working on a fix for this.

--- a/drivers/pwm/pwm_sam.c
+++ b/drivers/pwm/pwm_sam.c
@@ -107,7 +107,7 @@ static const struct pwm_driver_api sam_pwm_driver_api = {
 	};								\
 									\
 	DEVICE_DT_INST_DEFINE(inst,					\
-			    &sam_pwm_init, NULL				\
+			    &sam_pwm_init, NULL,			\
 			    NULL, &sam_pwm_config_##inst,		\
 			    POST_KERNEL,				\
 			    CONFIG_KERNEL_INIT_PRIORITY_DEVICE,		\

--- a/soc/arm/atmel_sam/common/pwm_fixup.h
+++ b/soc/arm/atmel_sam/common/pwm_fixup.h
@@ -1,0 +1,17 @@
+/*
+ * Copyright (c) 2021 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _ATMEL_SAM_PWM_FIXUP_H_
+#define _ATMEL_SAM_PWM_FIXUP_H_
+
+/* The SAMV71 HALs change the name of the field, so we need to
+ * define it this way to match how the other SoC variants name it
+ */
+#if defined(CONFIG_SOC_SERIES_SAMV71)
+#define PWM_CH_NUM PwmChNum
+#endif
+
+#endif /* _ATMEL_SAM_PWM_FIXUP_H_ */

--- a/soc/arm/atmel_sam/samv71/soc.h
+++ b/soc/arm/atmel_sam/samv71/soc.h
@@ -78,4 +78,6 @@
 
 #endif /* _ASMLANGUAGE */
 
+#include "pwm_fixup.h"
+
 #endif /* _ATMEL_SAMV71_SOC_H_ */

